plugin/core/views.py

```diff
@@ -808,3 +809,24 @@ def format_completion(
     if item.get("textEdit"):
         completion.flags = sublime.COMPLETION_FLAG_KEEP_PREFIX
     return completion
+
+
+def init_region_keys(view: sublime.View, server: str, doc_highlight_style: str) -> None:
+    doc_highlight_kinds = ["text", "read", "write"]
+    r = [sublime.Region(0, 0)]
+    for modifiers in SEMANTIC_SCOPES.values():
+        for scope in modifiers.values():
+            view.add_regions(scope, r)
+
+    if doc_highlight_style == 'fill':
+        for name in doc_highlight_kinds:
+            view.add_regions('lsp_highlight_{}{}'.format(name, 's'), r)
+            view.add_regions('lsp_highlight_{}{}'.format(name, 'm'), r)
+    for i in range(1, 5):
```

Comment:
I think these should be initialized in reversed order, so that errors are drawn over warnings and infos, if they span the same region.

There are also currently missing the region keys for diagnostics with tags from
https://github.com/sublimelsp/LSP/blob/a31b6b45301d612e69bc9a492fb75251b98bd393/plugin/session_view.py#L212-L213
and those should be initialized before the "regular" diagnostics.

***

I don't know why the documentHighlights would only work 50% of the time, maybe you can debug it by adding a print statement here and another one in https://github.com/sublimelsp/LSP/blob/a31b6b45301d612e69bc9a492fb75251b98bd393/plugin/documents.py#L607
to see if the documentHighlights request is somehow sent before `init_region_keys` was called?

Reply:
I've tried to debug it with print statements before, ` def render_highlights_on_main_thread() -> None: `, isn't getting called before `init_region_keys` but still sometimes doc_highlight refuses to draw.

Reply:
sometimes `def _clear_highlight_regions(self) -> None:` is getting called before `init_region_keys`,  in those cases , doc_highlighting  doesn't work. I tried to prevent 'def _clear_highlight_regions(self) -> None:' from getting called before `init_region_keys` but it didn't help.